### DISCUSSION TITLE
Why prisma starts an http server

### DISCUSSION THREAD

**Discussion by Yaojian** [prisma/prisma#6180] [Q&A] [answered]
When enable logging, the console shows:

```
prisma:info  Starting a sqlite pool with 21 connections.  
prisma:info  Started http server on http://127.0.0.1:13536
```

I am curious about why prisma starts a http server and what is the server for?

**Comment by ryands17** [comment, marked as answer]
@Yaojian 👋 

The http server is for Prisma's Node.js client to talk to the Query engine as shown [here](https://www.prisma.io/docs/concepts/components/prisma-client/query-engine#the-query-engine-at-runtime). This will be removed once [N-API](https://github.com/prisma/prisma/issues/4911) is implemented which you can read in this issue as well :)

**Reply by pantharshit00** [reply]
Also, small note, if you don't use to use TCP for engine communication, you can also use Unix Domain Sockets via this: https://github.com/prisma/prisma/issues/6088


But yes as Ryan mentioned this bridge is going to be removed as we are moving to NAPI based bridge which enables us to call Rust functions from node directly.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
